botocore/exceptions.py

```diff
@@ -52,6 +53,16 @@ class EndpointConnectionError(BotoCoreError):
         'Could not connect to the endpoint URL: "{endpoint_url}"')
 
 
+class BadStatusLineError(ConnectionError):
+    fmt = (
+        'Bad status line occurred on endpoint URL: "{endpoint_url}". '
```

Comment:
I think the intent was to completely hide anything about a bad status line from the customer.  This is confusing and misleading.  I think instead we should say something along the lines that it appears that the server closed the connection before we received a response.

We also can't really say you should increase retry attempts until we have a (supported) way for them to actually do so.